File: src/database/check_database.py
import json
import sys
import os

try:
    from db_manager import DatabaseManager
except ImportError:
    from database.db_manager import DatabaseManager

def check_database():
    try:
        db = DatabaseManager()
        all_coins = db.get_all_coins()
        
        # Get basic stats
        total_coins = len(all_coins)
        
        # Get the most recent coin
        most_recent = all_coins[-1] if all_coins else None
        
        # Format results as JSON
        result = {
            "status": "success",
            "total_coins": total_coins,
            "database_file": os.path.abspath("coins.db"),
            "most_recent": {
                "id": most_recent.id,
                "title": most_recent.title,
                "country": most_recent.country
            } if most_recent else None
        }
        
        print(json.dumps(result))
    except Exception as e:
        print(json.dumps({"status": "error", "message": str(e)}))

if __name__ == "__main__":
    check_database()

File: src/database/check_database.js
const path = require('path');
const db = require('./db.js');

function checkDatabase() {
    try {
        // Get all coins
        const allCoins = db.getAllCoins();
        
        // Get basic stats
        const totalCoins = allCoins.length;
        
        // Get the most recent coin
        const mostRecent = totalCoins > 0 ? allCoins[allCoins.length - 1] : null;
        
        // Format results as JSON
        const result = {
            status: "success",
            total_coins: totalCoins,
            database_file: path.resolve("coins.db"),
            most_recent: mostRecent ? {
                id: mostRecent.id,
                title: mostRecent.title,
                country: mostRecent.country
            } : null
        };
        
        console.log(JSON.stringify(result));
        return result;
    } catch (e) {
        const error = {
            status: "error", 
            message: e.message
        };
        console.error(JSON.stringify(error));
        return error;
    }
}

// Run the function if this script is executed directly
if (require.main === module) {
    checkDatabase();
}

// Export for use in other files
module.exports = checkDatabase;